python/ray/serve/tests/unit/test_deployment_state.py

```diff
@@ -332,22 +331,27 @@ def create_deployment_state_manager(
         dead_replicas_context.clear()
 
 
-def replica(version: Optional[DeploymentVersion] = None) -> VersionedReplica:
-    if version is None:
-        version = DeploymentVersion(get_random_string(), DeploymentConfig(), {})
+class FakeDeploymentReplica(DeploymentReplica):
```

Comment:
Nitpick: As a fake implementation we probably don't want it to inherit from the real `DeploymentReplica`. We should instead implement any method with the fake implementation if needed.

Reply:
We use the `VersionedReplica` type hint throughout the code, but outside of testing, the object is always a `DeploymentReplica`. It makes the type hints in the code harder to follow, and it makes `jump to definition` less useful.

Reply:
@edoakes @GeneDer what do you think?

Reply:
don't have that strong of an opinion. I prefer what we have now but if it's a practical inconvenience to you we can merge this

Reply:
+1 don't have strong opinion on taking `VersionedReplica` out. But maybe we should at least have `FakeDeploymentReplica` not inherit from `DeploymentReplica`

Reply:
Sounds good, I've removed the inheritance, and I'll merge this after it passes tests.